File: scraping/compare.py
import pandas as pd
import json

def run():
    # Read food.txt which represents the INDB extracts
    with open('food.txt', 'r', encoding='utf-8') as f:
        food_txt_lines = [l.strip() for l in f if l.strip()]
        
    # Clean the food txt items (remove commas, quotes)
    food_txt_clean = []
    for f in food_txt_lines:
        clean_name = f.strip('",').strip()
        if clean_name:
             food_txt_clean.append(clean_name)
             
    original_set = set()
    unique_food_txt = []
    for f in food_txt_clean:
        if f.lower() not in original_set:
            original_set.add(f.lower())
            unique_food_txt.append(f)

    # Read CSV
    df = pd.read_csv('myfitnesspal_nutrition_data.csv')
    searched = set(df['Search Term'].dropna().str.lower().str.strip())
    
    matches = [f for f in unique_food_txt if f.lower() in searched]
    remaining = [f for f in unique_food_txt if f.lower() not in searched]
    
    with open('remaining_foods.txt', 'w', encoding='utf-8') as f:
        f.write('\n'.join(remaining))
        
    print(f"Total Unique Foods from original list: {len(unique_food_txt)}")
    print(f"Total Matches Found in CSV: {len(matches)}")
    print(f"Remaining Foods (Not in CSV): {len(remaining)}")
    print("Wrote remaining foods to scraping/remaining_foods.txt")

if __name__ == "__main__":
    run()

File: scraping/compare.js
require('dotenv').config({ path: '../server/.env' });
const mongoose = require('mongoose');
const fs = require('fs');
const IndbFood = require('../server/models/IndbFood');
const csvParse = require('csv-parse/sync');

async function run() {
  try {
    await mongoose.connect(process.env.MONGO_URI);
    console.log('Connected to MongoDB');

    const foods = await IndbFood.find({}, 'displayName');
    const indbNamesRaw = foods.map(f => f.displayName || '');
    const indbNamesLow = indbNamesRaw.map(n => n.toLowerCase().trim());

    const csvContent = fs.readFileSync('myfitnesspal_nutrition_data.csv', 'utf8');
    const records = csvParse.parse(csvContent, { columns: true, skip_empty_lines: true });
    
    // Some CSVs enclose the food name field or have extra spaces.
    const csvNames = records.map(r => (r['Search Term'] || '').toLowerCase().trim());

    const matchingIndbItems = indbNamesRaw.filter(n => csvNames.includes(n.toLowerCase().trim()));
    const remainingIndbItems = indbNamesRaw.filter(n => !csvNames.includes(n.toLowerCase().trim()));

    fs.writeFileSync('matching_indb_foods.txt', matchingIndbItems.join('\n'));
    fs.writeFileSync('remaining_indb_foods.txt', remainingIndbItems.join('\n'));

    console.log('--- DB & CSV OVERLAP ---');
    console.log(`Total INDB foods in DB: ${indbNamesRaw.length}`);
    console.log(`Total searched foods in CSV: ${csvNames.length}`);
    console.log('');
    console.log(`Matched perfectly: ${matchingIndbItems.length}`);
    console.log(`Remaining in INDB not scraped: ${remainingIndbItems.length}`);
    console.log('Saved to "matching_indb_foods.txt" and "remaining_indb_foods.txt".');

    process.exit(0);
  } catch (err) {
    console.error(err);
    process.exit(1);
  }
}

run();